lang/csharp/src/apache/main/AvroDecimal.cs

```diff
@@ -0,0 +1,929 @@
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given float.
+        /// </summary>
+        /// <param name="value">The float value.</param>
+        public AvroDecimal(float value)
+            : this((decimal)value)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given decimal.
+        /// </summary>
+        /// <param name="value">The decimal value.</param>
+        public AvroDecimal(decimal value)
+        {
+            var bytes = GetBytesFromDecimal(value);
+
+            var unscaledValueBytes = new byte[13];
+            Array.Copy(bytes, unscaledValueBytes, unscaledValueBytes.Length - 1);
+
+            var unscaledValue = new BigInteger(unscaledValueBytes);
+            var scale = bytes[14];
+
+            if (bytes[15] == 128)
+                unscaledValue *= BigInteger.MinusOne;
+
+            UnscaledValue = unscaledValue;
+            Scale = scale;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given int.
+        /// </summary>
+        /// <param name="value">The int value.</param>
+        public AvroDecimal(int value)
+            : this(new BigInteger(value), 0) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given long.
+        /// </summary>
+        /// <param name="value">The long value.</param>
+        public AvroDecimal(long value)
+            : this(new BigInteger(value), 0) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given unsigned int.
+        /// </summary>
+        /// <param name="value">The unsigned int value.</param>
+        public AvroDecimal(uint value)
+            : this(new BigInteger(value), 0) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given unsigned long.
+        /// </summary>
+        /// <param name="value">The unsigned long value.</param>
+        public AvroDecimal(ulong value)
+            : this(new BigInteger(value), 0) { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given <see cref="BigInteger"/>
+        /// and a scale.
+        /// </summary>
+        /// <param name="unscaledValue">The double value.</param>
+        /// <param name="scale">The scale.</param>
+        public AvroDecimal(BigInteger unscaledValue, int scale)
+        {
+            UnscaledValue = unscaledValue;
+            Scale = scale;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AvroDecimal"/> class from a given array of bytes.
+        /// </summary>
+        /// <param name="value">The byte array.</param>
+        public AvroDecimal(byte[] value)
```

Comment:
this constructor unused and seems confusing to have in the public API, especially as what the bytes represent isn't documented.